### ISSUE TITLE
`set_upper_bound` does not seem to apply to all input variables

### ISSUE THREAD

**Issue by yenicelik** [adwhit/rust-nlopt#14]
When I do

```
opt.set_upper_bound(upper_bound).unwrap();
```

and later check the logs if the upper bounds were successfully set with 

```
println!("get_upper_bounds {:?}", opt.get_upper_bounds()); 
```

I get the following stdout, which is wrong: 

```
get_upper_bounds Some([1.6432184408e-313, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000])
```

This is also true for `set_upper_bounds`, as well as `set_lower_bound{s}`. As you can see, the first variable is not properly handled. Is there perhaps an off-by-one error when translating libraries perhaps (perhaps due to reference pointer when using vector vs constant-sized array)?

For reference, I am using the Sbplx algorithm. This is true for different values too.

**Comment by adwhit** [comment]
Not clear from your example what you expect to see? What value are you trying to set?

Quick glance at the bindings suggests maybe we should be using the `set_upper_bounds1` variant.

**Comment by adwhit** [comment]
Actually I see the problem, it is with the getter not the setter. It does something highly illegal (returns a dangling pointer)

**Comment by adwhit** [comment]
Fixed in `v0.7.0`
